dockerExtension.ts

```diff
@@ -166,13 +152,40 @@ export async function activate(ctx: vscode.ExtensionContext): Promise<void> {
     activateLanguageClient(ctx);
 }
 
-export async function deactivate(): Promise<void> {
+function registerContainerCommands(): void {
+    registerCommand('vscode-docker.container.start', startContainer);
+    registerCommand('vscode-docker.container.start.interactive', startContainerInteractive);
+    registerCommand('vscode-docker.container.start.azurecli', startAzureCLI);
+    registerCommand('vscode-docker.container.stop', stopContainer);
+    registerCommand('vscode-docker.container.restart', restartContainer);
+    registerCommand('vscode-docker.container.show-logs', showLogsContainer);
+    registerCommand('vscode-docker.container.open-shell', openShellContainer);
+    registerCommand('vscode-docker.container.remove', removeContainer);
+}
+
+function registerImageCommands(): void {
+    registerCommand('vscode-docker.image.build', buildImage);
+    registerCommand('vscode-docker.image.inspect', inspectImage);
+    registerCommand('vscode-docker.image.remove', removeImage);
+    registerCommand('vscode-docker.image.push', pushImage);
+    registerCommand('vscode-docker.image.tag', tagImage);
+
+}
+
+function registerComposeCommands(): void {
+    registerCommand('vscode-docker.compose.up', composeUp);
+    registerCommand('vscode-docker.compose.down', composeDown);
+    registerCommand('vscode-docker.compose.restart', composeRestart);
+    registerCommand('vscode-docker.system.prune', systemPrune);
+}
+
+export function deactivate(): Thenable<void> {
     if (!client) {
         return undefined;
     }
     // perform cleanup
     Configuration.dispose();
-    return await client.stop();
+    return client.stop();

```

Comment:
Why this change?

Reply:
The original function isn't async. I got an error regarding this. I'll admit this felt weird.

Reply:
You mean deactivate?  This is what I see:

export async function deactivate(): Promise<void> {
    if (!client) {
        return undefined;
    }
    // perform cleanup
    Configuration.dispose();
    return await client.stop();
}